x-pack/plugins/monitoring/server/lib/cluster/get_clusters_stats.ts

```diff
@@ -98,19 +110,17 @@ function fetchClusterStats(req: LegacyRequest, esIndexPattern: string, clusterUu
  * @param {Object} response The response from Elasticsearch.
  * @return {Array} Objects representing each cluster.
  */
-export function handleClusterStats(response: ElasticsearchResponse) {
+export function handleClusterStats(response: ElasticsearchResponse, req: LegacyRequest) {
   const hits = response?.hits?.hits ?? [];
 
   return hits
     .map((hit) => {
       const cluster = hit._source as ElasticsearchModifiedSource;
 
       if (cluster) {
-        const indexName = hit._index;
-        const ccs = parseCrossClusterPrefix(indexName);
-
         // use CCS whenever we come across it so that we can avoid talking to other monitoring clusters whenever possible
-        if (ccs) {
+        const ccs = getConfigCcs(req.server.config());
```

Comment:
So depending on the config, we can either have a boolean or a string that represents a cluster name ?
nit: we can cache this variable in the upper scope

Reply:
Yes

Reply:
I misunderstood what was happening here and undid these changes